xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +202,29 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
+        
+        // Exclude First Heading 
+        if (parameters.excludeFirstHeading()) {
+
+            // Getting the first block.
+            Block firstBlock = result.getChildren().get(0);
+
+            // Check if the first Block is Section block.
+            if (firstBlock instanceof SectionBlock) {
+                // Check if first block of section is Header Block.
+                Block sectionFirstBlock = firstBlock.getChildren().get(0);
+                if (sectionFirstBlock instanceof HeaderBlock) {
+                    // Removing the header block
+                    firstBlock.removeBlock(sectionFirstBlock);
```

Comment:
I don't think you need this. You just need to save the children of the HeaderBlock and to use  them in the replaceChild below.